gms/impl/src/main/java/com/linkedin/metadata/resources/entity/EntityResource.java

```diff
@@ -121,32 +129,51 @@
 
   @Action(name = ACTION_INGEST)
   @Nonnull
-  public Task<Void> ingest(@ActionParam(PARAM_ENTITY) @Nonnull Entity entity) throws URISyntaxException {
+  public Task<Void> ingest(
+      @ActionParam(PARAM_ENTITY) @Nonnull Entity entity,
+      @ActionParam(SYSTEM_METADATA) @Optional @Nullable SystemMetadata systemMetadata
+  ) throws URISyntaxException {
+
+    if (systemMetadata == null) {
+      SystemMetadata generatedSystemMetadata = new SystemMetadata();
+      generatedSystemMetadata.setLastObserved(System.currentTimeMillis());
+      generatedSystemMetadata.setRunId(DEFAULT_RUN_ID);
+      systemMetadata = generatedSystemMetadata;
+    }
+
     final Set<String> projectedAspects = new HashSet<>(Arrays.asList("browsePaths"));
     final RecordTemplate snapshotRecord = RecordUtils.getSelectedRecordTemplateFromUnion(entity.getValue());
     final Urn urn = com.linkedin.metadata.dao.utils.ModelUtils.getUrnFromSnapshot(snapshotRecord);
 
-    log.info("INGEST urn {}", urn.toString());
+    log.info("INGEST urn {} with system metadata {}", urn.toString(), systemMetadata.toString());
 
     final Entity browsePathEntity = _entityService.getEntity(urn, projectedAspects);
     BrowsePathUtils.addBrowsePathIfNotExists(entity.getValue(), browsePathEntity);
 
     // TODO Correctly audit ingestions.
     final AuditStamp auditStamp =
         new AuditStamp().setTime(_clock.millis()).setActor(Urn.createFromString(DEFAULT_ACTOR));
+    SystemMetadata finalSystemMetadata = systemMetadata;
```

Comment:
nit: Why do we need to do this?

Reply:
Java requires vars used in lambda expressions to be final or effectively final- adding the final flag and a comment so its more clear

Reply:
ah I didn't know it still worked without the final. awesome